aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1297,20 +1325,34 @@ AT_ERROR("svd: MAGMA library not found in "
   scalar_t wkopt;
   magma_int_t* iwork;
   ALLOCATE_ARRAY(iwork, magma_int_t, 8 * k);
-  magmaSvd<scalar_t>(jobz, m, n, self_data, m, S_data, U_data, m, VT_data, n, &wkopt, lwork, iwork, &info);
-  lwork = magma_int_cast(wkopt, "work_size");
+
+  value_t* rwork = nullptr;
+  if (isComplexType(at::typeMetaToScalarType(self.dtype()))) {
+    auto mx = std::max(m, n);
+    int64_t lrwork;
+    if (jobchar == 'N'){
+      lrwork = 7 * k;
+    }else if (mx > 10 * k){
```

Comment:
From [the MAGMA docs](https://icl.cs.utk.edu/projectsfiles/magma/doxygen/group__magma__gesdd.html), this condition should be `mx >= mn * 17/9`.